source/request-as-event-emitter.ts

```diff
@@ -235,37 +190,34 @@ export default (options: NormalizedOptions, input?: TransformStream) => {
 
 			request.on('error', onError);
 
-			timings = timer(request);
+			timings = timer(request); // TODO: Make `@szmarczak/http-timer` set `request.timings` and `response.timings`
 
+			const uploadBodySize = httpOptions.headers['content-length'] ? Number(httpOptions.headers['content-length']) : undefined;
 			uploadProgress(request, emitter, uploadBodySize);
 
-			if (options.timeout) {
-				timedOut(request, options.timeout, options);
-			}
+			timedOut(request, options.timeout, options.url);
 
 			emitter.emit('request', request);
 
+			if (isAborted()) {
+				return;
+			}
+
 			try {
-				if (is.nodeStream(options.body)) {
-					const {body} = options;
-					delete options.body;
-
-					// `stream.pipeline(…)` handles `error` for us.
-					request.removeListener('error', onError);
-
-					stream.pipeline(
-						body,
-						request,
-						uploadComplete
-					);
-				} else if (options.body) {
-					request.end(options.body, uploadComplete);
-				} else if (input && (options.method === 'POST' || options.method === 'PUT' || options.method === 'PATCH')) {
-					stream.pipeline(
-						input,
-						request,
-						uploadComplete
-					);
+				if (options.method === 'POST' || options.method === 'PUT' || options.method === 'PATCH') {
+					if (is.nodeStream(httpOptions.body)) {
+						// `stream.pipeline(…)` handles `error` for us.
+						request.removeListener('error', onError);
+
+						stream.pipeline(
+							// @ts-ignore TS is dumb.
```

Comment:
This is actually my fault: https://github.com/sindresorhus/is/commit/aeb3f74d653cf745abf2460f0896207de00c6059

Reply:
Ok, let me know when a new release is out :)